libpdbg/device.c

```diff
@@ -733,6 +733,11 @@ bool pdbg_targets_init(void *fdt)
 	dt_expand(pdbg_dt_root, dtb->system.fdt);
 
 	pdbg_targets_init_virtual(pdbg_dt_root, pdbg_dt_root);
+
+	//Before returning close any FD that might be still open
+	close(dtb->system.fd);
+	close(dtb->backend.fd);
```

Comment:
Line 730 checks if the backend fdt is valid and calls specific function. What if that is not valid, will the backend fd still be valid and can be closed? 

When we return using line no.728, are we sure the file is not opened? Because in that case, we are not closing it

Reply:
For LN730 case the fdt object is a FILE structure that by time is well populated. When we are returning at LN728 if the root node is not valid then the caller will restart all the processes as the minimum requirement is that the root node should be good after pdbg init call is returned.